src/studio/src/designer/frontend/packages/schema-editor/src/utils.ts

```diff
@@ -16,11 +16,27 @@ function flat(input: UiSchemaItem[] | undefined, depth = 1, stack: UiSchemaItem[
 }
 
 export function getUiSchemaItem(schema: UiSchemaItem[], path: string): UiSchemaItem {
-  // should we cache this flattened structure in state somehow for faster lookups?
-  const items = flat(schema, 999);
-  return items.find((i) => i.id === path) || {} as UiSchemaItem;
+  const matches = schema.filter((s) => path.includes(s.id));
+  if (matches.length === 1 && matches[0].id === path) {
+    return matches[0];
+  }
+  const items = flat(matches, 999);
+  const result = items.find((i) => i.id === path);
+  if (!result) {
+    throw new Error(`no uiSchema found: ${path}`);
+  }
+  return result;
 }
 
+export const getParentPath = (path: string): string | null => {
+  // eslint-disable-next-line no-useless-escape
+  if (path.match('[^#]\/properties')) {
```

Comment:
Why not use literal notation? Can then avoid the eslint disable statement.
```suggestion
  if (path.match(/[^#]\/properties/)) {
```